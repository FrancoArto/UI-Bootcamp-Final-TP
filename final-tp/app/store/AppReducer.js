import { combineReducers } from 'redux';
import tweetsReducer from './tweets/tweetsReducer'
import trendsReducer from './trends/trendsReducer'
import settingsReducer from './settings/settingsReducer'
import userReducer  from './users/userReducer'

const AppReducer = combineReducers({
  tweetsReducer,
  trendsReducer,
  settingsReducer,
  userReducer  
});

export default AppReducer;