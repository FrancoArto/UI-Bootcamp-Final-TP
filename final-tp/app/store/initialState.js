const tweetsInitialState = {
    data: [],
    loading: false,
    loadingSingleTweet: false,
    error: null,
    searchText: null,
    searchResults: [],
    userTimeline: [],
    singleTweetId: null,
    singleTweet: {}
};

const userInitialState = {
    loading: false,
    error: null,
    userId: null,
    userData: {}
}

const trendsInitialState = {
    data: [],
    loading: false,
    error: null,
}

//False is for 'not checked'
const appViewsConfigs = {
    verified: false,
    following: false,
    defaultInfo: false,
    withLink: false,
    withTruncatedText: false
}


export { tweetsInitialState, appViewsConfigs, userInitialState, trendsInitialState };