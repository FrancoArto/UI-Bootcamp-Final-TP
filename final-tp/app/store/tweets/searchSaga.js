import { GET_SEARCHTWEETS_URL, GET_APPENDRESULTS_URL } from '../../api/apiUrls';
import { takeLatest, put, call, select, all } from 'redux-saga/effects';
import { FETCH_SEARCH_BEGIN, fetchSearchSuccess, fetchSearchError, FETCH_MORE_RESULTS_REQUEST, fetchMoreResultsSuccess, fetchMoreResultsFailure } from './searchTweetsActions';

const getSearchText = state => state.searchTweetsReducer.searchText
const getMaxId = state => state.searchTweetsReducer.data[state.searchTweetsReducer.data.length - 1].id
const count = 50


function* fetchSearchResults() {  
  try {
    const searchText = yield select(getSearchText)
    const response = yield call(fetch, GET_SEARCHTWEETS_URL(searchText))
    const data = yield call([response, "json"]);
    yield put(fetchSearchSuccess(data.statuses))
  } catch (er) {
    yield put(fetchSearchError(er))
  }
}

function* searchTweets() {
  yield takeLatest(FETCH_SEARCH_BEGIN, fetchSearchResults)   
}


function* fetchResultsToAppend() {
  try {
    const searchText = yield select(getSearchText)
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDRESULTS_URL(searchText, count, maxId))
    const results = yield call([response, "json"]);
    const resultArray = results.statuses
    const data = resultArray.slice(1)
    yield put(fetchMoreResultsSuccess(data))
  } catch (er) {
    yield put(fetchMoreResultsFailure(er))
  }
}

function* appendResults() {
  yield takeLatest(FETCH_MORE_RESULTS_REQUEST, fetchResultsToAppend)   
}



export function* searchSaga() {
  yield all([
    searchTweets(),
    appendResults()
  ])
}