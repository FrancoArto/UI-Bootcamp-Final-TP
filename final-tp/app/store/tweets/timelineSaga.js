import { GET_TIMELINE_URL, GET_APPENDTWEETS_URL } from '../../api/apiUrls';
import { FETCH_TIMELINE_BEGIN, fetchTimelineSuccess, fetchTimelineFailure, FETCH_MORE_TWEETS_REQUEST, fetchMoreTweetsSuccess, fetchMoreTweetsFailure } from './timeLineActions'
import { takeLatest, put, call, all, select } from 'redux-saga/effects';

const twitTimelineCount = 50;

function* fetchTimeline() {
  try {
    const response = yield call(fetch, GET_TIMELINE_URL(twitTimelineCount))
    const data = yield call([response, "json"]);
    yield put(fetchTimelineSuccess(data))
  } catch (er) {
    yield put(fetchTimelineFailure(er))
  }
}

function* loadTimeline() {
  yield takeLatest(FETCH_TIMELINE_BEGIN, fetchTimeline)   
}

const getMaxId = state => state.timeLineReducer.data[state.timeLineReducer.data.length - 1].id


function* fetchTweetsToAppend() {
  try {
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDTWEETS_URL(twitTimelineCount, maxId))
    const array = yield call([response, "json"]);
    const data = array.slice(1)
    yield put(fetchMoreTweetsSuccess(data))
  } catch (er) {
    yield put(fetchMoreTweetsFailure(er))
  }
}

function* appendTweets() {
  yield takeLatest(FETCH_MORE_TWEETS_REQUEST, fetchTweetsToAppend)   
}

export function* timelineSaga() {
  yield all([
    loadTimeline(),
    appendTweets()
  ])
}