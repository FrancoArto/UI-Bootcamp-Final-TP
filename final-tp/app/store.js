import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga' 
import AppReducer from './reducers/index'; //Import the reducer

const sagaMiddleware = createSagaMiddleware()
 
export default createStore(AppReducer, applyMiddleware(sagaMiddleware));
