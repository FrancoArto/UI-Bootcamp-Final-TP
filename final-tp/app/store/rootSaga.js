import { all } from 'redux-saga/effects'
import {trendsSaga} from './trends/trendsSaga'
import { userSaga } from './users/userSaga';
import { tweetsSaga } from './tweets/tweetsSaga';

export default function* rootSaga() {
  yield all([
    tweetsSaga(),
    trendsSaga(),
    userSaga()
  ])
}
