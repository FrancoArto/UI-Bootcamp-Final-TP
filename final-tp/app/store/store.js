import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import AppReducer from '../reducers/index'; //Import the reducer
import rootSaga from '../sagas/index';
import reactotron from '../reactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export default reactotron.createStore(AppReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga)
