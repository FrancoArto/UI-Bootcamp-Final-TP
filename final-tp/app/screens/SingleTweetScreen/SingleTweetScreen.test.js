import React from 'react';
import {mount, shallow} from 'enzyme'
import ReduxSingleTweetScreen, {SingleTweetScreen} from './SingleTweetScreen';
import store from '../../store/store';
import { singleTweetData } from '../../store/tweets/tweetsMock';



it('renders without crashing', () => {
  const rendered = shallow(<SingleTweetScreen />)
  expect(rendered).toMatchSnapshot();
});

it('renders connected to store', () => {
  const rendered = shallow(<ReduxSingleTweetScreen store={store} />)
  expect(rendered).toMatchSnapshot();
});

it('renders activity indicator', () => {
  const rendered = shallow(<SingleTweetScreen
    tweet={singleTweetData}
    loading={true} />)
  expect(rendered).toMatchSnapshot();

  rendered.setState({
    loading: false
  })
  rendered.update()
  expect(rendered).toMatchSnapshot();
});

it('renders loaded tweet', () => {
  const rendered = shallow(<SingleTweetScreen
    tweet={singleTweetData}
    loading={false} />)
  expect(rendered).toMatchSnapshot();

  rendered.setState({
    loading: false
  })
  rendered.update()
  expect(rendered).toMatchSnapshot();
});



it('handles user profile link', () => {
  const navigation = {
    navigate: () => {}
  }

  const onUserProfilePress = jest.fn()
  
  const wrapper = mount(<SingleTweetScreen
    tweet={singleTweetData}
    loading={false}
    fetchUserDataRequest={onUserProfilePress}
    navigation={navigation} />)

  wrapper.setState({
    loading: false
  })
  wrapper.update()
  const tweet = wrapper.find('TouchableOpacity').first();
  tweet.props().onPress();
  expect(onUserProfilePress).toHaveBeenCalled()
});