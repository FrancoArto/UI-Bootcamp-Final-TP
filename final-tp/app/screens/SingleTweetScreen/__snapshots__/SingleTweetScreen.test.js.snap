// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders activity indicator 1`] = `<AppLoading />`;

exports[`renders activity indicator 2`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</Component>
`;

exports[`renders connected to store 1`] = `
<SingleTweetScreen
  fetchUserDataRequest={[Function]}
  loading={false}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tweet={Object {}}
/>
`;

exports[`renders loaded tweet 1`] = `<AppLoading />`;

exports[`renders loaded tweet 2`] = `
<Component
  style={
    Array [
      Object {
        "backgroundColor": "#f4f4f4",
        "flex": 1,
        "flexDirection": "column",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "borderWidth": 0,
          "flex": 0.5,
          "flexDirection": "row",
          "margin": 0,
          "paddingBottom": 0,
          "paddingTop": 12.5,
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "padding": 12.5,
            "width": 187.5,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          large={true}
          source={
            Object {
              "uri": "https://pbs.twimg.com/profile_images/1053387941752369152/r4pGBF3K_normal.jpg",
            }
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "marginRight": 250,
            "marginTop": 12.5,
            "padding": 12.5,
            "width": 562.5,
          },
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 0,
              "padding": 0,
            },
          ]
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#040404",
                "fontSize": 37.5,
                "fontWeight": "bold",
              },
            ]
          }
        >
           
          Clarín
           
        </Styled(Text)>
      </Component>
      <Component
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 0,
              "padding": 0,
            },
          ]
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#5D5D5D",
                "fontSize": 28.846153846153847,
              },
            ]
          }
        >
           @
          clarincom
        </Styled(Text)>
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "borderWidth": 0,
          "flex": 0.6,
          "flexDirection": "row",
          "marginBottom": 0,
          "marginLeft": 12.5,
          "marginRight": 12.5,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 12.5,
          "paddingRight": 12.5,
          "paddingTop": 0,
        },
      ]
    }
  >
    <_class
      linkDefault={true}
      linkStyle={
        Object {
          "color": "#2980b9",
        }
      }
    >
      <Styled(Text)
        style={
          Array [
            Object {
              "color": "#5D5D5D",
              "fontSize": 37.5,
            },
          ]
        }
      >
         
        Detuvieron a un cura en San Pedro: lo acusan de abusar de nenes de un jardín de infantes https://t.co/Cqm0vehy5P https://t.co/OLSi1XswSf
         
      </Styled(Text)>
    </_class>
  </Component>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 0,
          "paddingBottom": 9.375,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://pbs.twimg.com/media/DtqVRn_XgAAVAHD.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 20,
            "flex": 0.98,
            "height": "90%",
            "marginBottom": 0,
            "marginTop": "2%",
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flex": 0.5,
          "flexDirection": "row",
          "height": 222.33333333333334,
          "justifyContent": "center",
          "margin": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 9.375,
        },
      ]
    }
  >
    <Styled(Button)
      iconLeft={true}
      primary={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column-reverse",
            "margin": 0,
            "padding": 0,
          },
        ]
      }
      transparent={true}
    >
      <Styled(Icon)
        name="heart"
        style={
          Array [
            Object {
              "color": "#909090",
              "fontSize": 75,
              "marginLeft": 9.375,
              "marginRight": 9.375,
            },
          ]
        }
        type="SimpleLineIcons"
      />
      <Styled(Text)
        style={
          Array [
            Object {
              "color": "#5D5D5D",
              "fontSize": 41.666666666666664,
            },
          ]
        }
      >
        26
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Button)
      iconLeft={true}
      primary={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column-reverse",
            "margin": 0,
            "padding": 0,
          },
        ]
      }
      transparent={true}
    >
      <Styled(Icon)
        name="refresh"
        style={
          Array [
            Object {
              "color": "#909090",
              "fontSize": 75,
              "marginLeft": 9.375,
              "marginRight": 9.375,
            },
          ]
        }
        type="SimpleLineIcons"
      />
      <Styled(Text)
        style={
          Array [
            Object {
              "color": "#5D5D5D",
              "fontSize": 41.666666666666664,
            },
          ]
        }
      >
        14
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Button)
      iconLeft={true}
      primary={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column-reverse",
            "margin": 0,
            "padding": 0,
          },
        ]
      }
      transparent={true}
    >
      <Styled(Icon)
        name="bubble"
        style={
          Array [
            Object {
              "color": "#909090",
              "fontSize": 75,
              "marginLeft": 9.375,
              "marginRight": 9.375,
            },
          ]
        }
        type="SimpleLineIcons"
      />
      <Styled(Text)
        style={
          Array [
            Object {
              "color": "#5D5D5D",
              "fontSize": 41.666666666666664,
            },
          ]
        }
      />
    </Styled(Button)>
  </Component>
</Component>
`;

exports[`renders without crashing 1`] = `<AppLoading />`;
